release: v1.49.0 (#5418)
apollographql/router: .changesets/feat_bnjjj_feat_otel_name_attribute.md

### Override tracing span names using custom span selectors ([Issue #5261](https://github.com/apollographql/router/issues/5261))

> It gives you the ability to override the span name by using custom telemetry with any [selectors](https://www.apollographql.com/docs/router/configuration/telemetry/instrumentation/selectors/) you want just by setting the `otel.name` attribute.
> 
> Example:

Reviewer: Is this an accurate rewording?

```suggestion
Adds the ability to override span names by setting the `otel.name` attribute on any custom telemetry [selectors](https://www.apollographql.com/docs/router/configuration/telemetry/instrumentation/selectors/) .

This example changes the span name to `router`:
```

```yaml
telemetry:
  instrumentation:
    spans:
      router:
        otel.name:

>            static: router # Override the span name to router 

Reviewer: What is the original span name? (I thought it was `router` and am therefore confused.)

Author: @bnjjj Can you answer this outside of this PR?  I'll resolve this so we can merge the release.

Reviewer 2: It is router by default but if you have a GraphQL query with an operation name it's renamed with the operation name. This example is an example to show how you can enforce it to always keep router as span name
